### PR TITLE
Feature: Added PHPUnit

### PR DESCRIPTION
**Pull request by Martijnsvb**
Added PHPUnit to project and workflow.

Also added tests for the SharedConfig class but skipped the binaryConnect function. Since that protocol isn't supported anymore it feels a bit useless to write tests for it.

### CODE REVIEW

File: .github/workflows/php-tests.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ jobs:
         with:
           php-version: ${{ matrix.php-versions }}
           tools: cs2pr, phpcs, phpstan
+          extensions: uopz
```
Comment (SMillerDev) [inline .github/workflows/php-tests.yml:25]: https://github.com/marketplace/actions/setup-php-action#pcov should probably define this too

File: .github/workflows/php-tests.yml [hunk 2]
```diff
@@ -43,3 +44,6 @@ jobs:
 
       - name: Run sample
         run: php sample_push.php | grep "Trying tls://gateway.sandbox.push.apple.com:2195"
+
+      - name: Run PHPUnit
```
Comment (SMillerDev) [inline .github/workflows/php-tests.yml:48]: https://github.com/marketplace/actions/setup-php-action#phpunit this needs to be defined before

**Review by SMillerDev — COMMENTED** [review]
Some tweaks, what is the coverage level for the project and for SharedConfig after this?

File: .github/workflows/php-tests.yml [hunk 3]
```diff
@@ -43,3 +49,11 @@ jobs:
 
       - name: Run sample
         run: php sample_push.php | grep "Trying tls://gateway.sandbox.push.apple.com:2195"
+
+      - name: Setup PCOV
+        run: |
+          composer require pcov/clobber
+          vendor/bin/pcov clobber
+
```
Comment (SMillerDev) [inline .github/workflows/php-tests.yml:57]: > If you are using PHPUnit 5.x, 6.x or 7.x, you need to set up pcov/clobber before executing your tests.

We're using 9.x so:
```suggestion
```

Comment (Martijnsvb) [reply .github/workflows/php-tests.yml:57]: Yea I see that now. I thought it said php 8. Reading is hard...

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**